python/ray/tune/execution/experiment_state.py

```diff
@@ -210,38 +214,15 @@ def checkpoint(
             wait: Wait until sync to cloud has finished.
 
         """
-        experiment_local_path = self._storage.experiment_local_path
-        if not experiment_local_path:
-            return
-
         force = force or self._should_force_cloud_sync
+        if force:
+            self._process_launcher.launch(get_save_fn())
+        else:
+            if self._process_launcher.ready_for_next_launch:
+                self._process_launcher.launch(get_save_fn())
 
-        now = time.time()
-        if now - self._last_save_time < self._checkpoint_period and not force:
-            return
-
-        # Checkpoint
-        checkpoint_time_start = time.monotonic()
-
-        # NOTE: This context manager is for Datasets captured in a trial config.
-        # This is the case when *tuning over datasets*.
-        # If the datasets have already been full executed, then serializing
-        # block refs means that this checkpoint is not usable in a new Ray cluster.
-        # This context will serialize the dataset execution plan instead, if available.
-        with out_of_band_serialize_dataset():
-            save_fn()
-
-        # Sync to cloud
-        self.sync_up(force=force, wait=wait)
-
-        checkpoint_time_taken = time.monotonic() - checkpoint_time_start
-
-        # Adjust dynamic checkpointing
-        self._update_auto_checkpoint_time(time_taken=checkpoint_time_taken)
```

Comment:
This "auto-detected" experiment checkpoint period no longer really applies.

There used to be 2 checkpoint periods:
1. "auto checkpoint time", which is how many seconds between each call to `save_fn`, which *writes files to disk.*
2. `sync_period` in `sync_up`, which controls how many seconds between each upload to storage.

Now, there's no more "writing files to disk" (number 1) -- we only need to configure the sync up period. The default of every 5 minutes may be too infrequent -- we need to guard against ungraceful experiment termination.